api/v1beta2/network_types.go

```diff
@@ -1002,3 +1027,57 @@ func (z ZoneType) String() string {
 func (z ZoneType) Equal(other ZoneType) bool {
 	return z == other
 }
+
+// ElasticIPPool defines the Elastic IP pool alows custom configuration for resources allocating
+// public IPv4 addresses on public subnets.
+type ElasticIPPool struct {
+	// PublicIpv4Pool set a custom Public IPv4 Pool used to created Elastic IP address for resources
+	// created in public IPv4 subnets. Every IPv4 address, Elastic IP, will be allocated from the custom
+	// Public IPv4 pool that you brought to AWS, instead of Amazon-provided pool.
+	//
+	// +kubebuilder:validation:MaxLength=30
```

Comment:
I didn't found anything from AWS documentation, this limitation is based in our development based in couple of pools we are allocating while developing the support on OpenShift. The "pool ID" have all the same length. I will need to take more research to try to find something official :)

Reply:
According to the AWS SDK docs (https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#CreatePublicIpv4PoolOutput), the pool ID is generated by AWS and returned to the user when they create a pool. I can't find much about the length of these IDs either, but if it's generated by AWS, it's possible that their algorithm for making these IDs will change in the future.

That said, I doubt it will change much.

Reply:
@nrb thanks for update. So are we find keeping `MaxLength=30`?

Reply:
Yes, let's keep max length at 30.